docs/testing/unit-testing.md

```diff
@@ -0,0 +1,406 @@
+
+- Because the Office JavaScript APIs library isn't loaded in the node process, the `Office` object that is referenced in the add-in code must be declared and initialized.
+
+```javascript
+const OfficeMockObject = require("office-addin-mock");
+const myOutlookAddinFeature = require("../my-outlook-add-in-feature");
+
+// Create the seed mock object.
+const mockData = {
+  context: {
+    mailbox: {
+      item: {
+          setSelectedDataAsync: function (data) {
+          this.data = data;
+        },
+      },
+    },
+  },
+};
+  
+// Create the final mock object from the seed object.
+const officeMock = new OfficeMockObject.OfficeMockObject(mockData);
+
+// Create the Office object that is called in the addHelloWorldText function.
+global.Office = officeMock;
+
+/* Code that calls the test framework goes below this line. */
+
+// Jest test
+test("Text of selection in message should be set to 'Hello World'", async function () {
+    await myOutlookAddinFeature.addHelloWorldText();
+    expect(officeMock.context.mailbox.item.data).toBe("Hello World!");
+});
+```
+
+### Mocking the Office application-specific APIs
+
+When you are testing functions that use the application-specific APIs, be sure that you are mocking the right type of object. There are two options:
+
+- Mock a [OfficeExtension.ClientRequestObject](/javascript/api/office/officeextension.clientrequestcontext). Do this when the function that is being tested meets both of the following conditions:
+
+  - It doesn't call a *Host*.`run` method, such as [Excel.run](/javascript/api/excel#Excel_run_batch_).
+  - It doesn't reference any other direct property or method of a *Host* object.
+
+- Mock a *Host* object, such as [Excel](/javascript/api/excel) or [Word](/javascript/api/word). Do this when the preceding option isn't possible.
+
+Examples of both types of tests are in the subsections below.
+
+#### Mocking a ClientRequestContext object
+
+This example assumes an Excel add-in that has one of its features in a file named `my-excel-add-in-feature.js`. The following shows the contents of the file. Note that the `getSelectedRangeAddress` is a helper method that is called inside the callback that is passed to `Excel.run`.
+
+```javascript
+const myExcelAddinFeature = {
+    
+    getSelectedRangeAddress: async (context) => {
+        const range = context.workbook.getSelectedRange();      
+        range.load("address");
+
+        await context.sync();
+      
+        return range.address;
+    }
+}
+
+module.exports = myExcelAddinFeature;
+```
+
+The test file, named `my-excel-add-in-feature.test.js` is in a subfolder, relative to the location of the add-in code file. The following shows the contents of the file. Note that the top level property is `workbook`, so the object that is being mocked is the parent of an `Excel.workbook`: a `ClientRequestContext` object.
```

Comment:
I believe "an `Excel.workbook`" should either be "an `Excel.Workbook`" (if we're referring to the object type) or "`Excel.workbook`" (if we're referring to the host-object property).